tensorflow_lite_support/c/task/vision/image_classifier.h

```diff
@@ -0,0 +1,189 @@
+// the score threshold are returned. The value of score threshold
+// should be between 0 and 1.
+extern void ImageClassifierOptionsSetScoreThreshold(
+    ImageClassifierOptions* image_classifier_options, float score_threshold);
+
+// Sets the number of threads used for classification in the encapsulated 
+// ImageClassifierOptions
+extern void ImageClassifierOptionsSetNumThreads(
+    ImageClassifierOptions* image_classifier_options, int num_threads);
+
+// Adds a class name to the list of class names to be whitelisted. If you have
+// more than one class names to be whitelisted, consider calling this method
+// repeatedly in a loop. If you set atleast one class name into the whitelist 
+// using this option, classifications whose class names which are not added 
+// into the white list will be filtered out. Duplicate or unknown
+// class names are ignored. Mutually exclusive with blacklisted class names set
+// using void ImageClassifierOptionsAddClassNameBlackList(
+// ImageClassifierOptions* image_classifier_options, char* class_name);
+extern void ImageClassifierOptionsAddClassNameWhiteList(
+    ImageClassifierOptions* image_classifier_options, char* class_name);
+
+// Adds a class name to the list of class names to be black listed. If you have
+// more than one class names to be black listed, consider calling this method
+// repeatedly in a loop. Classifications whose class are added into the black 
+// list will be filtered out. Duplicate or unknown class names are ignored. 
+// Mutually exclusive with white listed class names set
+// using void ImageClassifierOptionsAddClassNameWhiteList(
+// ImageClassifierOptions* image_classifier_options, char* class_name);
+extern void ImageClassifierOptionsAddClassNameBlackList(
+    ImageClassifierOptions* image_classifier_options, char* class_name);
+
+//Disposes off the ImageClassifierOptions.
+void ImageClassifierOptionsDelete(ImageClassifierOptions* options);
+
+
+extern const struct ImageClassifierOptions ImageClassifierOptions_default;
+
+// Creates ImageClassifier from options. When using this function to instantiate
+// Image Classifer, the model path should be set using
+// ImageClassifierOptionsSetModelFilePath(ImageClassifierOptions* options, 
+//                                          const char* model_path)
+// returns nullptr if the file doesn't exist or is not a well formatted 
+// TFLite model path.
+extern ImageClassifier* ImageClassifierFromOptions(
+    const ImageClassifierOptions* options);
+
+// Creates ImaegeClassifier from model path and default options, returns nullptr
+// if the file doesn't exist or is not a well formatted TFLite model path.
+// This image classifier created using this function defaults to s score 
+// threshold of 0, max_results of 5. In order to override these, use:
+// ImageClassifier* ImageClassifierFromOptions(
+//   const ImageClassifierOptions* options) instead.
+extern ImageClassifier* ImageClassifierFromFile(const char* model_path);
```

Comment:
ImageClassifierFromFile is not needed once you integrate BaseOptions in ImageClassifierOptions.

Reply:
Fixed